src/extractor/guidance/sliproad_handler.cpp

```diff
@@ -29,273 +32,720 @@ SliproadHandler::SliproadHandler(const IntersectionGenerator &intersection_gener
-                    candidate_in = intersection[1].eid;
-                    intersection = intersection_generator(node, candidate_in);
-                }
-                return intersection;
-            }(intersection_node_id);
-
-            const auto link_data = node_based_graph.GetEdgeData(road.eid);
-            // Check if the road continues here
-            const bool is_through_street =
-                !target_intersection.empty() && link_data.name_id != EMPTY_NAMEID &&
-                target_intersection.end() !=
-                    std::find_if(
-                        target_intersection.begin() + 1,
-                        target_intersection.end(),
-                        [this, &link_data](const ConnectedRoad &road) {
-                            const auto &road_edge_data = node_based_graph.GetEdgeData(road.eid);
-
-                            const auto same_name =
-                                road_edge_data.name_id != EMPTY_NAMEID &&
-                                !util::guidance::requiresNameAnnounced(road_edge_data.name_id,
-                                                                       link_data.name_id,
-                                                                       name_table,
-                                                                       street_name_suffix_table);
-
-                            return same_name;
-                        });
-
-            // if the sliproad candidate is a through street, we cannot handle it as a sliproad
-            if (is_through_street)
+            continue;
+        }
+
+        // If the sliproad candidate is a through street, we cannot handle it as a sliproad.
+        if (isThroughStreet(sliproad_edge, target_intersection))
+        {
+            continue;
+        }
+
+        // The turn off of the Sliproad has to be obvious and a narrow turn
+        {
+            const auto index = findObviousTurn(sliproad_edge, target_intersection);
+
+            if (index == 0)
+                continue;
+
+            const auto onto = target_intersection[index];
+            const auto angle_deviation = angularDeviation(onto.angle, STRAIGHT_ANGLE);
+            const auto is_narrow_turn = angle_deviation <= 2 * NARROW_TURN_ANGLE;
+
+            if (!is_narrow_turn)
+                continue;
+        }
+
+        // Check for curvature. Depending on the turn's direction at `c`. Scenario for right turn:
+        //
+        // a ... b .... c .   a ... b .... c .   a ... b .... c .
+        //       `      .           `  .   .             .    .
+        //         `    .                . .           .      .
+        //           `  .                 ..             .    .
+        //              d                  d                . d
+        //
+        //                    Sliproad           Not a Sliproad
+        {
+            // Intersection is orderd: 0 is UTurn, then from sharp right to sharp left.
```

Comment:
some of these checks are quite expensive. Have you checked whether you can re-order some of them so that cheaper ones come first to exit early when possible?

Can you potentially move some of these into dedicated functions (detectors) and annotate them in `O(n)`. That could make finding a reasonable order easier.